src/hotspot/share/opto/vectorization.cpp

```diff
@@ -1783,3 +1785,417 @@ void AlignmentSolver::trace_constrained_solution(const int C_const,
+  // This was just wrapped. Now we simply unwap without touching the inputs.
+  return VTransformApplyStatus::make_scalar(_node);
+}
+
+VTransformApplyStatus VTransformReplicateNode::apply(const VLoopAnalyzer& vloop_analyzer, const GrowableArray<Node*>& vnode_idx_to_transformed_node) const {
+  Node* val = find_transformed_input(1, vnode_idx_to_transformed_node);
+  VectorNode* vn = VectorNode::scalar2vector(val, _vlen, _element_type);
+  register_new_node_from_vectorization(vloop_analyzer, vn, val);
+  return VTransformApplyStatus::make_vector(vn, _vlen, vn->length_in_bytes());
+}
+
+VTransformApplyStatus VTransformConvI2LNode::apply(const VLoopAnalyzer& vloop_analyzer, const GrowableArray<Node*>& vnode_idx_to_transformed_node) const {
+  Node* val = find_transformed_input(1, vnode_idx_to_transformed_node);
+  Node* n = new ConvI2LNode(val);
+  register_new_node_from_vectorization(vloop_analyzer, n, val);
+  return VTransformApplyStatus::make_scalar(n);
+}
+
+VTransformApplyStatus VTransformShiftCountNode::apply(const VLoopAnalyzer& vloop_analyzer, const GrowableArray<Node*>& vnode_idx_to_transformed_node) const {
+  PhaseIdealLoop* phase = vloop_analyzer.vloop().phase();
+  Node* shift_count_in = find_transformed_input(1, vnode_idx_to_transformed_node);
+  assert(shift_count_in->bottom_type()->isa_int(), "int type only for shift count");
+  // The shift_count_in would be automatically truncated to the lowest _mask
+  // bits in a scalar shift operation. But vector shift does not truncate, so
+  // we must apply the mask now.
+  Node* shift_count_masked = new AndINode(shift_count_in, phase->igvn().intcon(_mask));
+  register_new_node_from_vectorization(vloop_analyzer, shift_count_masked, shift_count_in);
+  // Now that masked value is "boadcast" (some platforms only set the lowest element).
+  VectorNode* vn = VectorNode::shift_count(_shift_opcode, shift_count_masked, _vlen, _element_bt);
+  register_new_node_from_vectorization(vloop_analyzer, vn, shift_count_in);
+  return VTransformApplyStatus::make_vector(vn, _vlen, vn->length_in_bytes());
+}
+
+
+VTransformApplyStatus VTransformPopulateIndexNode::apply(const VLoopAnalyzer& vloop_analyzer, const GrowableArray<Node*>& vnode_idx_to_transformed_node) const {
+  PhaseIdealLoop* phase = vloop_analyzer.vloop().phase();
+  Node* val = find_transformed_input(1, vnode_idx_to_transformed_node);
+  assert(val->is_Phi(), "expected to be iv");
+  assert(VectorNode::is_populate_index_supported(_element_bt), "should support");
+  const TypeVect* vt = TypeVect::make(_element_bt, _vlen);
+  VectorNode* vn = new PopulateIndexNode(val, phase->igvn().intcon(1), vt);
+  register_new_node_from_vectorization(vloop_analyzer, vn, val);
+  return VTransformApplyStatus::make_vector(vn, _vlen, vn->length_in_bytes());
+}
+
+VTransformApplyStatus VTransformElementWiseVectorNode::apply(const VLoopAnalyzer& vloop_analyzer, const GrowableArray<Node*>& vnode_idx_to_transformed_node) const {
```

Comment:
Note: in the future, I could split up this `VTransformElementWiseVectorNode` into a number of sub-classes and thus avoid all the case distinctions below. But for now I keep it a bit closer to the old `SuperWord::apply_vectorization` with its case distinctions. There is also a lot of shared code between the cases, so I don't want to blow up the code size too much for now.